### ISSUE TITLE
Cannot update via "Update Warp" on Arch Linux - Keyserver failure

### ISSUE THREAD

**Issue by tscalia74** [warpdotdev/Warp#4312]
### Describe the bug

When attempting to update via the "Update Warp" button on Arch Linux, update fails with the following error:

gpg: error retrieving 'linux-maintainers@warp.dev' via WKD: No data
gpg: error retrieving 'linux-maintainers@warp.dev' via keyserver: Server indicated a failure
gpg: error reading key: Server indicated a failure
==> ERROR: Remote key not fetched correctly from WKD or keyserver.

### To reproduce

1. Open Warp Termina
2. Click on "Update Warp" button in upper right corner
3. Result:
![Screenshot from 2024-02-24 09-40-04](https://github.com/warpdotdev/Warp/assets/121516073/522f5f0c-08e0-4e4e-9f97-36066ded1d3b)
4. Press enter, 
5. Type sudo password and press enter again
6. Result
![Screenshot from 2024-02-24 09-42-46](https://github.com/warpdotdev/Warp/assets/121516073/1555bc34-a88c-4487-ba91-cda8512c80ad)

### Expected behavior

The program update would proceed

### Operating system

Linux

### Operating system and version

Arch (rolling - fully up to date)

### Shell Version

bash 5.2.26

### Current Warp version

v0.2024.02.20.08.01.stable_01v0.2024.02.20.08.01.stable_01

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by LFd3v** [comment]
It seems to be related to [DNS resolution](https://unix.stackexchange.com/questions/399027/gpg-keyserver-receive-failed-server-indicated-a-failure) for some package versions.

Here I use a custom DNS via `dnscrypt-proxy`, and the command seen in the update also fails, but either of these two work:

    sudo pacman-key -r "linux-maintainers@warp.dev" --keyserver hkp://keys.openpgp.org:80

or (let the system pick/resolve the key server):

    sudo pacman-key -r "linux-maintainers@warp.dev"

@tscalia74 If you haven't updated yet, probably using the instructions in the i[nstallation guide](https://docs.warp.dev/getting-started/getting-started-with-warp#arch-linux-based-distributions) will allow you to update.

**Comment by vorporeal** [comment]
@LFd3v Thanks for pointing out that `hkp://keys.openpgp.org:80` is more robust than simply `keys.openpgp.org`; going to change the suggested command to use that instead!

**Comment by eliyya** [comment]
I have the same problem, but apart from the fact that warp does not update, now no package can be updated or installed

After the failed `Click on "Update Warp" button`, every time I try to do some operation like `yay -Syyu` or a simple `yay -Sy spotify` I get the error: `error: warpdotdev: signature from "Warp Linux Maintainers (Package Signing Authority) <linux-maintainers@warp.dev>" is unknown trust`
![image](https://github.com/warpdotdev/Warp/assets/62972439/c11c6a52-739f-40a7-9721-f5b5b3d8ea62)

When I tried an old solution from a forum (`sudo pacman -U https://archive.archlinux.org/packages/a/archlinux-keyring/...`) just to try to see a different error I ran into the error `database 'warpdotdev' is not valid`

That's when it occurred to me to edit `/etc/pacman.conf` and remove the warpdotdev repo
![image](https://github.com/warpdotdev/Warp/assets/62972439/ae70ad8a-6134-45eb-97d9-8b1989aa2456)

After that it was back to normal
![image](https://github.com/warpdotdev/Warp/assets/62972439/5f24e9d4-24d5-4589-9305-ddbcd90ec556)

I wasn't sure if this goes in a separate issue or as it comes from this problem it goes here.

For what it's worth I use `Archcraft 6.7.2-arch1-1` and use warp version `v0.2024.02.20.08.01.stable_02`

**Comment by LFd3v** [comment]
@eliyya Yes, the problem was a result of adding the repo to pacman.conf but not having added the signing key. In other words, it was a result of the error during the update. This can be fixed by adding the key manually with one of the commands in my previous reply, or removing the repo as you did.

Cheers

**Comment by tscalia74** [comment]
The repository config was interrupted by the key server error, o clear
those errors, you'll need to open '/etc/pacman.conf' (as root or sudo),
scroll to the bottom, and delete the repository entry for warp.

Then run 'sudo pacman -Syyu'

**Comment by vorporeal** [comment]
@eliyya If you want to get Warp updates (recommended!), I'd suggest uncommenting the repository configuration in `/etc/pacman.conf` and adding our signing key:

```
sudo pacman-key -r "linux-maintainers@warp.dev" --keyserver "hkp://keys.openpgp.org:80"
sudo pacman-key --lsign-key "linux-maintainers@warp.dev"
```

Apologies that the command got your machine into a bad state!  I'm going to reorder the steps in repository initialization to ensure this doesn't happen!

**Comment by MRParag95** [comment]
sudo pacman-key -r "linux-maintainers@warp.dev" --keyserver "hkp://keys.openpgp.org:80"

gpg: error retrieving 'linux-maintainers@warp.dev' via WKD: No data
gpg: error retrieving 'linux-maintainers@warp.dev' via keyserver: No keyserver available
gpg: error reading key: No keyserver available
==> ERROR: Remote key not fetched correctly from WKD or keyserver.

**Comment by vorporeal** [comment]
Does it work if you remove the `--keyserver` argument and let it choose a key server?

**Comment by PrimotionStudio** [comment]
`sudo sh -c "echo -e '\n[warpdotdev]\nServer = https://releases.warp.dev/linux/pacman/\$repo/\$arch' >> /etc/pacman.conf"
sudo pacman-key -r "linux-maintainers@warp.dev" --keyserver hkp://keys.openpgp.org:80
sudo pacman-key --lsign-key "linux-maintainers@warp.dev"
sudo pacman -Sy warp-terminal`

This just solved it for me.

**Comment by PrimotionStudio** [comment]
/sudo sh -c "echo -e '\n[warpdotdev]\nServer = https://releases.warp.dev/linux/pacman/\$repo/\$arch' >> /etc/pacman.conf"
sudo pacman-key -r "linux-maintainers@warp.dev" --keyserver hkp://keys.openpgp.org:80
sudo pacman-key --lsign-key "linux-maintainers@warp.dev"
sudo pacman -Sy warp-terminal
